File: frontend/src/types/card.ts
export interface Card {
  id: number
  user_id: number
  original_filename: string
  storage_path: string
  ocr_text?: string
  ocr_confidence?: number
  extracted_name?: string
  extracted_company?: string
  extracted_phone?: string
  extracted_email?: string
  extracted_address?: string
  is_processed: boolean
  is_reviewed: boolean
  created_at: string
  updated_at?: string
}

File: backend/app/schemas/card.py
"""Card schemas."""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, EmailStr, ConfigDict


class CardBase(BaseModel):
    """Base card schema."""

    original_filename: str


class CardCreate(CardBase):
    """Card creation schema."""

    storage_path: str
    file_size: int


class CardUpdate(BaseModel):
    """Card update schema."""

    extracted_name: Optional[str] = None
    extracted_company: Optional[str] = None
    extracted_phone: Optional[str] = None
    extracted_email: Optional[EmailStr] = None
    extracted_address: Optional[str] = None
    is_reviewed: Optional[bool] = None


class CardResponse(CardBase):
    """Card response schema."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    storage_path: str
    ocr_text: Optional[str] = None
    ocr_confidence: Optional[float] = None
    extracted_name: Optional[str] = None
    extracted_company: Optional[str] = None
    extracted_phone: Optional[str] = None
    extracted_email: Optional[str] = None
    extracted_address: Optional[str] = None
    is_processed: bool
    is_reviewed: bool
    created_at: datetime
    updated_at: Optional[datetime] = None
